lib/pure/math.nim

```diff
@@ -184,25 +186,12 @@ proc signbit*(x: SomeFloat): bool {.inline, since: (1, 5, 1).} =
 
 func copySign*[T: SomeFloat](x, y: T): T {.inline, since: (1, 5, 1).} =
   ## Returns a value with the magnitude of `x` and the sign of `y`;
-  ## this works even if x or y are NaN or zero, both of which can carry a sign.
+  ## this works even if x or y are NaN, infinity or zero, all of which can carry a sign.
   runnableExamples:
-    doAssert copySign(1.0, -0.0) == -1.0
-    doAssert copySign(0.0, -0.0) == -0.0
-    doAssert copySign(-1.0, 0.0) == 1.0
-    doAssert copySign(10.0, 0.0) == 10.0
-
-    doAssert copySign(Inf, -1.0) == -Inf
```

Comment:
edge cases matter but you can also argue that the documentation already covers those edge cases as currently written. At least add:
`doAssert copySign(-Inf, -0.0) == -Inf`, and `doAssert copySign(NaN, 1.0).isNaN`